Makefile

```diff
@@ -83,6 +83,13 @@ deps: librime data
 ifdef ARCHS
 BUILD_SETTINGS += ARCHS="$(ARCHS)"
 BUILD_SETTINGS += ONLY_ACTIVE_ARCH=NO
+_=$() $()
+export CMAKE_OSX_ARCHITECTURES = $(subst $(_),;,$(ARCHS))
+endif
+
+ifdef MACOSX_DEPLOYMENT_TARGET
```

Comment:
if we specify this directly in the make command
`make MACOSX_DEPLOYMENT_TARGET=x.y` 
can we remove this ifdef ... endif block all together?

Reply:
The `BUILD_SETTINGS += ...` part is needed.
The export can be removed.

Reply:
I just wanted to create a shortcut, but it's fine to use the complete argument name

Reply:
Fine.
To specify this option, user has to know the details anyway.
Let's follow the recommended build target by xcode in upcoming Squirrel releases.
Wwith [this change](https://github.com/lotem/librime/commit/477c010efd9f6887aee7d0412f39ba4a326a6bc3) in librime, we won't need to specify `MACOSX_DEPLOYMENT_TARGET` in the make command.

Reply:
And one more thing, Sparkle need to be built from source if we also want the target OS also apply to Sparkle (there's a difference in size I've noticed at least)

Reply:
I wonder what happens if a library like Sparkle or `librime.dylib` was build for a lower target OS version than Squirrel.app?

Reply:
In my previous builds, I didn't change Sparkle's target(which defaults to 10.9). It results in a larger binary, but seems to work.

However, syncing the deployment targets of components makes peace of mind for me